codingStandards/CodingStandard-Java.md

```diff
@@ -550,9 +550,39 @@ This ensures that variables are valid at any time. Sometimes it is impossible to
 
 The concept of Java information hiding and encapsulation is violated by public variables. Use private variables and access functions instead. One exception to this rule is when the class is essentially a data structure, with no behavior (*equivalent to a C++ struct*). In this case it is appropriate to make the class' instance variables public.
 
+**9. `this.` keyword should not be used unless necessary.**
+
+Use the `this.` keyword only if the name of the class field is the same as the name of the method parameters (i.e. the class field is overshadowed by a method parameter).
+
+<table>
+  <tr>
+    <th align="center">Good</th>
+    <th align="center">Bad</th>
+  </tr>
+  <tr>
+    <td>
+      <pre lang="java">
+public User(String name) {
+    this.name = name;
+    ...
+}
+      </pre>
+    </td>
+    <td>
+      <pre lang="java">
+public User(String name) {
+    // 'id' is not overshadowed by any method parameters
+    this.id = User.getNewId();
+    ...
+}
+      </pre>
+    </td>
+  </tr>
+</table>
+
 ### **Loops**
 
-**9. The loop body should be wrapped by curly brackets irrespective of how many lines there are in the body**
+**10. The loop body should be wrapped by curly brackets irrespective of how many lines there are in the body**
```

Comment:
Created issue here: https://github.com/oss-generic/process/issues/70 
Further discussion can proceed there.
In the meantime, this manual numbering is inevitable and should be passed.